powerauth-backend-tests/src/test/java/com/wultra/security/powerauth/test/v31/PowerAuthApiTest.java

```diff
@@ -493,36 +494,44 @@ void callbackTest() throws PowerAuthClientException {
-        byte[] serverPublicKeyBytes = BaseEncoding.base64().decode(JsonUtil.stringValue(config.getResultStatusObjectV31(), "serverPublicKey"));
-        final ECPublicKey serverPublicKey = (ECPublicKey) keyConvertor.convertBytesToPublicKey(serverPublicKeyBytes);
-        final EciesEncryptor eciesEncryptor = eciesFactory.getEciesEncryptorForActivation(serverPublicKey, config.getApplicationSecret().getBytes(StandardCharsets.UTF_8),
-                transportMasterKeyBytes, EciesSharedInfo1.CREATE_TOKEN);
-        final EciesCryptogram eciesCryptogram = eciesEncryptor.encryptRequest("{}".getBytes(StandardCharsets.UTF_8), true);
-        EciesEncryptedRequest eciesRequest = new EciesEncryptedRequest();
-        eciesRequest.setEphemeralPublicKey(BaseEncoding.base64().encode(eciesCryptogram.getEphemeralPublicKey()));
-        eciesRequest.setEncryptedData(BaseEncoding.base64().encode(eciesCryptogram.getEncryptedData()));
-        eciesRequest.setMac(BaseEncoding.base64().encode(eciesCryptogram.getMac()));
-        eciesRequest.setNonce(BaseEncoding.base64().encode(eciesCryptogram.getNonce()));
-        CreateTokenResponse tokenResponse = powerAuthClient.createToken(config.getActivationIdV31(), config.getApplicationKey(), eciesRequest.getEphemeralPublicKey(), eciesRequest.getEncryptedData(),
-                eciesRequest.getMac(), eciesRequest.getNonce(), SignatureType.POSSESSION_KNOWLEDGE);
-        assertNotNull(tokenResponse.getEncryptedData());
-        assertNotNull(tokenResponse.getMac());
-        EciesCryptogram responseCryptogram = new EciesCryptogram(BaseEncoding.base64().decode(tokenResponse.getMac()), BaseEncoding.base64().decode(tokenResponse.getEncryptedData()));
-        byte[] decryptedData = eciesEncryptor.decryptResponse(responseCryptogram);
-        TokenResponsePayload response = objectMapper.readValue(decryptedData, TokenResponsePayload.class);
-        assertNotNull(response.getTokenId());
-        assertNotNull(response.getTokenSecret());
-        BaseStepModel model = new BaseStepModel();
-        model.setResultStatusObject(config.getResultStatusObjectV31());
-        CounterUtil.incrementCounter(model);
-        final byte[] tokenNonce = tokenGenerator.generateTokenNonce();
-        final byte[] tokenTimestamp = tokenGenerator.generateTokenTimestamp();
-        final byte[] tokenDigest = tokenGenerator.computeTokenDigest(tokenNonce, tokenTimestamp, BaseEncoding.base64().decode(response.getTokenSecret()));
-        ValidateTokenResponse validateResponse = powerAuthClient.validateToken(response.getTokenId(), BaseEncoding.base64().encode(tokenNonce), Long.parseLong(new String(tokenTimestamp)), BaseEncoding.base64().encode(tokenDigest));
+    void createValidateAndRemoveTokenTestActiveActivation() throws InvalidKeySpecException, CryptoProviderException, GenericCryptoException, IOException, EciesException, PowerAuthClientException {
```

Comment:
I prefer to keep it as is, because it's good to know what exceptions can occur in individual tests because we use a lot of cryptography in these tests because we are missing the integration layer.